consensus/polybft/bridge_event_manager.go

```diff
@@ -0,0 +1,569 @@
+				if numberOfMessages > 0 {
+					b.logger.Debug("can not submit a batch, quorum not reached",
+						"from", pendingBatch.BridgeMessageBatch.Messages[0].ID.Uint64(),
+						"to", pendingBatch.BridgeMessageBatch.Messages[numberOfMessages-1].ID.Uint64())
+				}
+
+				continue
+			}
+
+			return nil, err
+		}
+
+		largestBridgeBatch = &BridgeBatchSigned{
+			MessageBatch: pendingBatch.BridgeMessageBatch,
+			AggSignature: aggregatedSignature,
+		}
+
+		break
+	}
+
+	return largestBridgeBatch, nil
+}
+
+// getAggSignatureForBridgeBatchMessage checks if pending batch has quorum,
+// and if it does, aggregates the signatures
+func (b *bridgeEventManager) getAggSignatureForBridgeBatchMessage(blockNumber uint64,
+	pendingBridgeBatch *PendingBridgeBatch) (Signature, error) {
+	validatorSet := b.validatorSet
+
+	validatorAddrToIndex := make(map[string]int, validatorSet.Len())
+	validatorsMetadata := validatorSet.Accounts()
+
+	for i, validator := range validatorsMetadata {
+		validatorAddrToIndex[validator.Address.String()] = i
+	}
+
+	bridgeBatchHash, err := pendingBridgeBatch.Hash()
+	if err != nil {
+		return Signature{}, err
+	}
+
+	// get all the votes from the database for batch
+	votes, err := b.state.BridgeMessageStore.getMessageVotes(
+		pendingBridgeBatch.Epoch,
+		bridgeBatchHash.Bytes(),
+		pendingBridgeBatch.BridgeMessageBatch.DestinationChainID.Uint64())
+	if err != nil {
+		return Signature{}, err
+	}
+
+	var signatures bls.Signatures
+
+	publicKeys := make([][]byte, 0)
+	bmap := bitmap.Bitmap{}
+	signers := make(map[types.Address]struct{}, 0)
+
+	for _, vote := range votes {
+		index, exists := validatorAddrToIndex[vote.From]
+		if !exists {
+			continue // don't count this vote, because it does not belong to validator
+		}
+
+		signature, err := bls.UnmarshalSignature(vote.Signature)
+		if err != nil {
+			return Signature{}, err
+		}
+
+		bmap.Set(uint64(index))
+
+		signatures = append(signatures, signature)
+		publicKeys = append(publicKeys, validatorsMetadata[index].BlsKey.Marshal())
+		signers[types.StringToAddress(vote.From)] = struct{}{}
+	}
+
+	if !validatorSet.HasQuorum(blockNumber, signers) {
+		return Signature{}, errQuorumNotReached
+	}
+
+	aggregatedSignature, err := signatures.Aggregate().Marshal()
+	if err != nil {
+		return Signature{}, err
+	}
+
+	result := Signature{
+		AggregatedSignature: aggregatedSignature,
+		Bitmap:              bmap,
+	}
+
+	return result, nil
+}
+
+// PostEpoch notifies the bridge event manager that an epoch has changed,
+// so that it can discard any previous epoch bridge batch, and build a new one (since validator set changed)
+func (b *bridgeEventManager) PostEpoch(req *PostEpochRequest) error {
+	b.lock.Lock()
+
+	b.pendingBridgeBatches = nil
+	b.validatorSet = req.ValidatorSet
+	b.epoch = req.NewEpochID
+
+	// build a new batch at the end of the epoch
+	nextBridgeEventIDIndex, err := req.SystemState.GetNextCommittedIndex(b.sourceChainID)
```

Comment:
As a parameter we use b.sourceChainID and in the method itself a parameter is named destinationChainID. What is correct?